contracts/src/v0.8/ccip/interfaces/IMessageTransformer.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {Internal} from "../libraries/Internal.sol";
+
+/// @notice Interface for plug-in message hook contracts that transform OffRamp & OnRamp messages.
+///         The transformer functions are expected to revert on transform failures.
+interface IMessageTransformer {
+  /// @notice Common error that can be thrown on transform failures and used by consumers
+  /// @param errorReason abi encoded revert reason
+  error MessageTransformError(bytes errorReason);
+
+  /// @notice Transforms the given OffRamp message. Reverts on transform failure
+  /// @param message to transform
+  /// @return transformed message
+  function transformInboundMessage(
```

Comment:
nit switch the two function to follow the msg lifecycle.

Can you add something to the comments why this is useful or what type of transformations could exist?

Reply:
yes added